### PR TITLE
Spread extra props in components.

### LINKED ISSUE #2376: Spread extra props into components for event tracking.
**Is your feature request related to a problem? Please describe.**

Rather than having custom logic in psammead for simorgh event tracking, can we instead spread props into this component, and set the data attribute in simorgh please? IMO we shouldnt have custom logic specific to simorgh when its not needed. Someone else using Brand may not do event tracking like we are. It also makes it very difficult for us to change our event tracking method in the future.

**Describe the solution you'd like**
Revert work done on 
- https://github.com/bbc/psammead/pull/2354
- https://github.com/bbc/psammead/pull/2353
- https://github.com/bbc/psammead/pull/2343

Psammead component:
```js
export const IndexAlsos = props => (
    <div {...props}>
      {props.children}
    </div>
  );
```

Simorgh:
```js
  return (
    <IndexAlsos data-e2e="index-alsos">{...}</IndexAlsos>
```

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by ghost**
Resolves #2376

**Overall change:** _Spread extra props to component to keep simorgh logic on event tracking out of psammead._

**Code changes:**

- _Spread extra props to components to allow extra props like `data-*` attributes to be passed in simorgh._
  - `SectionLabel`
  - `NavigationLi`
  - `Brand`
  - `StoryPromo`
- Updated snapshot test and added assertion tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-navigation/src/index.jsx [hunk 1]
```diff
@@ -170,19 +171,14 @@ export const NavigationLi = ({
         script={script}
         currentLink="true"
         service={service}
-        data-navigation={`${link}_${url}`}
```
Comment (dr3) [inline packages/components/psammead-navigation/src/index.jsx:173]: This IMO makes this a breaking change, up to you wether you follow it, as it’s very minor

File: packages/components/psammead-section-label/src/index.jsx [hunk 2]
```diff
@@ -71,11 +71,9 @@ const SectionLabel = ({
   script,
   service,
   visuallyHidden,
+  ...props
 }) => (
-  <SectionLabelWrapper
-    visuallyHidden={visuallyHidden}
-    data-section-divider={labelId}
```
Comment (dr3) [inline packages/components/psammead-section-label/src/index.jsx:77]: As is this package :)

File: packages/components/psammead-story-promo/src/index.jsx [hunk 3]
```diff
@@ -311,8 +311,8 @@ LiveLabel.defaultProps = {
   dir: 'ltr',
 };
 
-const StoryPromo = ({ image, info, mediaIndicator, topStory, sectionName }) => (
-  <StoryPromoWrapper topStory={topStory} data-story-promo={sectionName}>
```
Comment (dr3) [inline packages/components/psammead-story-promo/src/index.jsx:315]: And this one :)

File: packages/components/psammead-navigation/src/index.jsx (hunk 1 above)
Comment (ghost) [reply packages/components/psammead-navigation/src/index.jsx:173]: Done. Bumped the versions

File: packages/components/psammead-navigation/src/index.test.jsx [hunk 4]
```diff
@@ -21,6 +22,7 @@ describe('Navigation', () => {
               active={active}
               currentPageText="Current page"
               service="news"
+              data-naviation="test_navigation"
```
Comment (DenisHdz) [inline packages/components/psammead-navigation/src/index.test.jsx]:
```suggestion
              data-navigation="test_navigation"
```

File: packages/components/psammead-brand/src/index.jsx [hunk 5]
```diff
@@ -174,6 +174,7 @@ const Brand = props => {
     borderBottom,
     backgroundColour,
     logoColour,
+    ...rest
```
Comment (DenisHdz) [inline packages/components/psammead-brand/src/index.jsx:177]: Is it `...rest` the same as `...props`? If so, can we keep the same naming across files?

**Review by DenisHdz — COMMENTED** [review]
Just a couple of things

File: packages/components/psammead-brand/src/index.jsx (hunk 5 above)
Comment (DenisHdz) [reply packages/components/psammead-brand/src/index.jsx:177]: Ok, the reason for naming it `...rest` here is because `props` is already used.

**Comment by tochwill** [comment]
Just noticed that packages with breaking changes are being bumped to new minor versions - if following semver should these be major versions?

**Comment by DenisHdz** [comment]
> Just noticed that packages with breaking changes are being bumped to new minor versions - if following semver should these be major versions?

Done :)

**Review by tochwill — APPROVED** [review]

File: packages/components/psammead-brand/src/index.jsx (hunk 5 above)
Comment (12) [reply packages/components/psammead-brand/src/index.jsx:177]: I'd have called it `additionalProps` or something, but that's up to you.

File: packages/components/psammead-brand/CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 | ------- | ----------- |
+| 5.0.6 | [PR#2377](https://github.com/bbc/psammead/pull/2377) Spread extra props to the component |
```
Comment (12) [inline packages/components/psammead-brand/CHANGELOG.md:6]: Should this be a breaking change as well, considering the others are?

Comment (DenisHdz) [reply packages/components/psammead-brand/CHANGELOG.md:6]: The patch should be fine since we are not removing props as in the others.

**Review by ryanmccombe — APPROVED** [review]

**Review by AlistairGempf — APPROVED** [review]
All looks good.

**Review by dr3 — APPROVED** [review]

**Pull request merged**